bdk_rpc_wallet_example/src/rpc.rs

```diff
@@ -0,0 +1,349 @@
+            sparse_chain.apply_changeset(changeset);
+
+            if current_hash == existing_hash {
+                break;
+            }
+        }
+
+        // Insert the new tip so new transactions will be accepted into the sparse chain.
+        let tip = self.get_tip().map(|(height, hash)| BlockId {
+            height: height as u32,
+            hash,
+        })?;
+        if let Err(failure) = sparse_chain.insert_checkpoint(tip) {
+            match failure {
+                sparse_chain::InsertCheckpointError::HashNotMatching { .. } => {
+                    // There has been a re-org before we even begin scanning addresses.
+                    // Just recursively call (this should never happen).
+                    return self.wallet_scan(local_chain);
+                }
+            }
+        }
+
+        let last_scanned_height = local_chain
+            .iter()
+            .rev()
+            .next()
+            .map_or(0, |(height, _)| *height as usize);
+
+        // Fetch the transactions
+        let page_size = 1000; // Core has 1000 page size limit
+        let mut page = 0;
+
+        let _ = self.rescan_blockchain(Some(last_scanned_height), None);
+
+        let mut txids_to_update = Vec::new();
+
+        loop {
+            // TODO: List transaction should take in a label.
+            let list_tx_result = self
+                .list_transactions(None, Some(page_size), Some(page * page_size), Some(true))?
+                .iter()
+                .filter(|item|
+                    // filter out conflicting transactions - only accept transactions that are already
+                    // confirmed, or exists in mempool
+                    item.info.confirmations > 0 || self.get_mempool_entry(&item.info.txid).is_ok())
+                .map(|list_result| {
+                    let chain_index = match list_result.info.blockheight {
+                        Some(height) => TxHeight::Confirmed(height),
+                        None => TxHeight::Unconfirmed,
+                    };
+                    (chain_index, list_result.info.txid)
+                })
+                .collect::<HashSet<(TxHeight, Txid)>>();
+
+            txids_to_update.extend(list_tx_result.iter());
+
+            if list_tx_result.len() < page_size {
+                break;
+            }
+            page += 1;
+        }
+
+        for (index, txid) in txids_to_update {
+            if let Err(failure) = sparse_chain.insert_tx(txid, index) {
+                match failure {
+                    sparse_chain::InsertTxError::TxTooHigh { .. } => {
+                        unreachable!("We should not encounter this error as we ensured tx_height <= tip.height");
+                    }
+                    sparse_chain::InsertTxError::TxMovedUnexpectedly { .. } => {
+                        /* This means there is a reorg, we will handle this situation below */
```

Comment:
It seems the situation is not handled. Maybe it is better to break here?

Reply:
I think the break happens in the next segment `// Check for Reorg during the above sync process`. I do think we should do better than just error in such cases. But this is how it's done for other examples, so I followed the same here too. 

A few lines above, when we first check for `Reorg during sync`, we call the scan function recursively. Maybe we can do the same for the case after extracting the list of transactions? i.e, if there's again a reorg during sync, call scan again? 

Open to suggestions here.

Reply:
Reorg during sync is a very low probability situation, and when the user gets this error, they should just run the scan again. In that perspective, it makes sense to call the scan again internally rather than throwing the error?